crates/core/src/db/datastore/locking_tx_datastore/mut_tx.rs

```diff
@@ -1229,6 +1235,31 @@ impl<'a> Iterator for IndexScanFilterDeleted<'a> {
 }
 
 impl MutTxId {
+    pub(crate) fn insert_st_client(&mut self, identity: Identity, connection_id: ConnectionId) -> Result<()> {
+        let row = &StClientRow {
+            identity: identity.into(),
+            connection_id: connection_id.into(),
+        };
+        self.insert_via_serialize_bsatn(ST_CLIENT_ID, row).map(|_| ())
+    }
+
+    pub(crate) fn delete_st_client(&mut self, identity: Identity, connection_id: ConnectionId) -> Result<()> {
+        let row = &StClientRow {
+            identity: identity.into(),
+            connection_id: connection_id.into(),
+        };
+        let ptr = self
+            .iter_by_col_eq(
+                ST_CLIENT_ID,
+                col_list![StClientFields::Identity, StClientFields::ConnectionId],
+                &AlgebraicValue::product(row),
+            )?
+            .next()
+            .expect("the client should be connected")
```

Comment:
Can we not just make the method a no-op in this case?

Reply:
I don't think that's necessarily true, actually: it's possible, though inadvisable, for the module owner to use `spacetime sql` to manually delete `st_client` rows. I think that we generally do not expect manually mutating the system tables to be well-behaved, but we don't want any customer to be able to kill our threads, which panicking might do. I am unsure if there is a `catch_unwind` above this in the call stack.